### ISSUE TITLE
After pinning a couple of hidden columns, visible columns cannot be pinned

### ISSUE THREAD

**Issue by sstoyanovIG** [IgniteUI/igniteui-angular#2531]
### Description  
I guess the hidden columns' width is taken into account for the max width of pinned columns.

 * igniteui-angular version: 6.1.3
 * browser: all

### Steps to reproduce  

1. Open https://staging.infragistics.local/products/ignite-ui-angular/angular/components/grid_toolbar.html#features
2. Hide 'Athlete' and 'Country' columns
3. Pin 'Athlete' and 'Country' columns
4. Try to pin another column

### Result  
No column can be pinned

### Expected result  
Columns should be able to be pinned.

**Comment by MayaKirova** [comment]
@sstoyanovIG By Design hidden columns are taken into account when calculating the min allowed width after which pinning is refused.

If hidden columns are not taken into account, showing them may put the grid in an invalid state (fixed cols may exceed the visible area so no scrollbar will be displayed).

There are different approaches in which we could handle this in future versions (however they all require coupling Hiding and Pinning), for example:
-	Prevent showing a previously hidden and pinned column if the total pinned width exceeds the allowed width.
-  Unpin a previously hidden and pinned column if the total pinned width exceeds the allowed width.
-	Prevent pinning hidden columns in the first place.

@ChronosSF @mpavlinov  Let me know what you think.

**Comment by ChronosSF** [comment]
Imo it doesn't make sense to show hidden columns in the pinning dropdown in the first place so the third option from Maya's list is what I'd go for. It should also be the easiest to implement.

**Comment by mpavlinov** [comment]
@MayaKirova @sstoyanovIG As the check is at run-time I think that the operation to show a hidden and pinned column should be canceled. @ChronosSF @rkaraivanov @kdinev What do you think?

**Comment by kdinev** [comment]
I have a question. What is the point of pinning a hidden column?

**Comment by mpavlinov** [comment]
@kdinev There is no point. but the UI allows it, so the end-user can do it.

**Comment by mpavlinov** [comment]
@kdinev I think we should close it. What do you think?

**Comment by kdinev** [comment]
@mpavlinov Does the UI still allow pinning hidden columns?

**Comment by mpavlinov** [comment]
@kdinev Yes. Here is the [sample](https://stackblitz.com/angular/glkmjbojdkv) for 7.2.0

**Comment by ChronosSF** [comment]
@kdinev , @radomirchev , please check @MayaKirova 's comment again.

Removing the hidden columns from the pinning menu won't be enough. Unless we disallow the properties hidden and pinned to be true at once through the API, the UI will have to be a lot more specific about the columns' states for the end-user to make sense of why trying to pin one doesn't work. 

What we could do but it'll not be as simple as a bug fix is disable pinned columns in the hiding dropdown and hidden columns in the pinning dropdown (removing them entirely will leave hidden+pinned columns out of the UI entirely, which may be desired but that's up for debate).

Then we know that the user can click on any of the hiding dropdown columns without breaking the grid state. If we have such "disable" functionality we can also apply it for the pinned dropdown to disable unpinned columns when the pinned area is full (and of course disable all hidden columns).

In either case, solutions to this issue require coupling the two collections in some way. If we don't want this, we can leave the current implementation as is.

**Comment by MayaKirova** [comment]
@ChronosSF @radomirchev @kdinev As discussed, we will remove all handling for scenarios where the user can force the grid into an invalid/broken state . We will allow him to pin/unpin or hide/show anything, regardless of the result.
